demos/main/src/main/java/com/google/android/exoplayer2/demo/SampleChooserActivity.java

```diff
@@ -239,6 +242,31 @@ public boolean onChildClick(
     intent.putExtra(
         IntentUtil.PREFER_EXTENSION_DECODERS_EXTRA,
         isNonNullAndChecked(preferExtensionDecodersMenuItem));
+    if (playlistHolder.mediaItems.size() == 1) {
+      AtomicInteger appendCount = new AtomicInteger(0);
+      Uri originUri = playlistHolder.mediaItems.get(0).localConfiguration.uri;
+      PlaybackProvider.getInstance().getStreamInfo(originUri)
+          .observeOn(AndroidSchedulers.mainThread())
+          .subscribe(result -> {
```

Comment:
I think we could simply the logic by removing the `onComplete`. 
If we return an `Observable.error(..)` in `PlaybackProvider#getStreamInfo` if the media is null, then we will have only two cases:

**onNext:**
```
IntentUtil.addToIntent(Collections.singletonList(result), intent);
startActivity(intent);
```

**onError:**
```
IntentUtil.addToIntent(playlistHolder.mediaItems, intent);
startActivity(intent);
```

Reply:
For the case of media is null, now it return empty `MediaItem` instance (It is not an error) and then we also can remove the `onComplete()`.